tests/admin.test.ts

```diff
@@ -182,3 +185,180 @@ describe('bonus points submission', () => {
+    users[1].accessType = UserAccessType.STANDARD;
+    users[2].accessType = UserAccessType.MARKETING;
+    users[3].accessType = UserAccessType.MERCH_STORE_DISTRIBUTOR;
+
+    await new PortalState()
+      .createUsers(users[0], users[1], users[2], users[3], admin)
+      .write();
+
+    const adminController = ControllerFactory.admin(conn);
+
+    const accessLevelResponse = await adminController.updateUserAccessLevel({
+      "accessUpdates": [
+        { "user": users[0].email,  "newAccess": {
+            "accessType": "MERCH_STORE_MANAGER"
+        }},
+        { "user": users[1].email,  "newAccess": {
+            "accessType": "MARKETING"
+        }},
+        { "user": users[2].email,  "newAccess": {
+            "accessType": "MERCH_STORE_DISTRIBUTOR"
+        }},
+        { "user": users[3].email,  "newAccess": {
+            "accessType": "STAFF"
+        }}
+      ]
+    }, admin);
+
+    // expect statements
+    expect(accessLevelResponse.updatedUsers[0].email).toEqual(users[0].email);
+    expect(accessLevelResponse.updatedUsers[0].accessType).toEqual(UserAccessType.MERCH_STORE_MANAGER);
+
+    expect(accessLevelResponse.updatedUsers[1].email).toEqual(users[1].email);
+    expect(accessLevelResponse.updatedUsers[1].accessType).toEqual(UserAccessType.MARKETING);
+
+    expect(accessLevelResponse.updatedUsers[2].email).toEqual(users[2].email);
+    expect(accessLevelResponse.updatedUsers[2].accessType).toEqual(UserAccessType.MERCH_STORE_DISTRIBUTOR);
+
+    expect(accessLevelResponse.updatedUsers[3].email).toEqual(users[3].email);
+    expect(accessLevelResponse.updatedUsers[3].accessType).toEqual(UserAccessType.STAFF);
+  });
+
+
+
+  // attempt to update when user is not an admin
+  test('attempt to update when user is not an admin', async () => {
+    const conn = await DatabaseConnection.get();
+    const standard = UserFactory.fake({ accessType: UserAccessType.STANDARD });
+    const users = UserFactory.create(5);
+
+
+    users[0].accessType = UserAccessType.STAFF;
+    users[1].accessType = UserAccessType.STANDARD;
+    users[2].accessType = UserAccessType.MARKETING;
+    users[3].accessType = UserAccessType.MERCH_STORE_DISTRIBUTOR;
+
+    await new PortalState()
+      .createUsers(users[0], users[1], users[2], users[3], standard)
+      .write();
+
+    const adminController = ControllerFactory.admin(conn);
+
+    await expect(async () => {
+      await adminController.updateUserAccessLevel({
+        "accessUpdates": [
+          { "user": users[0].email, "newAccess": { "accessType": "MERCH_STORE_MANAGER" } },
+          { "user": users[1].email, "newAccess": { "accessType": "MARKETING" } },
+          { "user": users[2].email, "newAccess": { "accessType": "MERCH_STORE_DISTRIBUTOR" } },
+          { "user": users[3].email, "newAccess": { "accessType": "STAFF" } }
+        ]
+      }, standard);
+    }).rejects.toThrow(ForbiddenError);
+  });
+
+  test('attempt to update duplicate users', async () => {
```

Comment:
@dowhep This test fails with a `QueryFailedError: duplicate key value violates unique constraint "Users_email_key"` error message, but works in Postman. If we update a user twice with two different access levels, it updates the user to the last instance of that access level (eg if we update someone to STAFF and then MARKETING, the final result will be marketing. What's the criteria for this issue coming up, and how do you suggest we handle it? My thought was to use a set to keep all emails unique.

Reply:
Although it is interesting that postman behaves differently than jest tests, will keep an eye on this and why that is. (last time this happened is due to a decorator typo so make sure all the decorators are correct)